api/leaderboard.go

```diff
@@ -10,842 +10,761 @@
+func (app *App) RemoveMembers(ctx context.Context, in *api.RemoveMembersRequest) (*api.RemoveMembersResponse, error) {
+	lg := app.Logger.With(
+		zap.String("handler", "RemoveMembers"),
+		zap.String("leaderboard", in.LeaderboardId),
+	)
+
+	if in.Ids == "" {
+		app.AddError()
+		return nil, status.New(codes.InvalidArgument,
+			"Member IDs are required using the 'ids' querystring parameter").Err()
 	}
-}
 
-//RemoveMembersHandler removes several members from a leaderboard
-func RemoveMembersHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		leaderboardID := c.Param("leaderboardID")
-		lg := app.Logger.With(
-			zap.String("handler", "RemoveMembersHandler"),
-			zap.String("leaderboard", leaderboardID),
-		)
+	memberIDs := strings.Split(in.Ids, ",")
+	idsInter := make([]interface{}, len(memberIDs))
+	for i, v := range memberIDs {
+		idsInter[i] = v
+	}
 
-		ids := c.QueryParam("ids")
-		if ids == "" {
+	err := withSegment("Model", ctx, func() error {
+		lg.Debug("Removing members.", zap.String("ids", in.Ids))
+		err := app.Leaderboards.RemoveMembers(ctx, in.LeaderboardId, idsInter)
+
+		if err != nil && !strings.HasPrefix(err.Error(), notFoundError) {
+			lg.Error("Members removal failed.", zap.Error(err))
 			app.AddError()
-			return FailWith(400, "Member IDs are required using the 'ids' querystring parameter", c)
+			return err
 		}
+		lg.Debug("Members removal succeeded.")
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
 
-		memberIDs := strings.Split(ids, ",")
-		idsInter := make([]interface{}, len(memberIDs))
-		for i, v := range memberIDs {
-			idsInter[i] = v
-		}
+	return &api.RemoveMembersResponse{Success: true}, nil
+}
 
-		err := WithSegment("Model", c, func() error {
-			lg.Debug("Removing members.", zap.String("ids", ids))
-			err := app.Leaderboards.RemoveMembers(c.StdContext(), leaderboardID, idsInter)
+// GetMember is the handler responsible for retrieving a member score and rank
+func (app *App) GetMember(ctx context.Context, in *api.GetMemberRequest) (*api.DefaultMemberResponse, error) {
+	lg := app.Logger.With(
+		zap.String("handler", "GetMember"),
+		zap.String("leaderboard", in.LeaderboardId),
+		zap.String("memberPublicID", in.MemberPublicId),
+	)
+
+	order := in.Order
+	if order == "" || (order != "asc" && order != "desc") {
+		order = "desc"
+	}
 
-			if err != nil && !strings.HasPrefix(err.Error(), notFoundError) {
-				lg.Error("Members removal failed.", zap.Error(err))
-				app.AddError()
-				return err
-			}
-			lg.Debug("Members removal succeeded.")
-			return nil
-		})
-		if err != nil {
-			return FailWith(500, err.Error(), c)
+	var member *leaderboard.Member
+	err := withSegment("Model", ctx, func() error {
+		var err error
+		lg.Debug("Getting member.")
+		member, err = app.Leaderboards.GetMember(ctx, in.LeaderboardId, in.MemberPublicId, order, in.ScoreTTL)
+		if err != nil && strings.HasPrefix(err.Error(), notFoundError) {
```

Comment:
nit: consider using a switch here.

Suggestion: it seems we'd benefit from a NotFound error in the lib. Consider adding it (in a separate PR).

Reply:
Added TODO for the change in the lib.